### PR TITLE
Add array flatten Presto scalar function

### PR DESCRIPTION
**Pull request by zacw7**
Flattens an array(array(T)) to an array(T) by concatenating the contained arrays.

### CODE REVIEW

**Review by amitkdutta — COMMENTED** [review]
@zacw7 We need to provide specialization for primitive types (e.g. https://github.com/facebookincubator/velox/pull/5221)

File: velox/functions/prestosql/ArrayFunctions.h [hunk 1]
```diff
@@ -554,4 +554,34 @@ struct ArrayTrimFunctionString {
...
+      elementCount += array.value().size();
+    }
+    out.reserve(elementCount);
+    for (const auto& array : arrays) {
+      for (auto element : array.value()) {
+        if (element.has_value()) {
+          auto& newItem = out.add_item();
+          newItem.copy_from(element.value());
+        } else {
+          out.add_null();
```
Comment (amitkdutta) [inline velox/functions/prestosql/ArrayFunctions.h:581]: This is only one layer? Not recursive flatten?

Comment (zacw7) [reply velox/functions/prestosql/ArrayFunctions.h:581]: Yes only one layer:

Run the following query in Presto:
```
SELECT flatten(ARRAY[ARRAY[ARRAY[1, 1], ARRAY[2, 2]], ARRAY[ARRAY[1, 1], ARRAY[2, 2]]])
```

Result:
```
[[1, 1], [2, 2], [1, 1], [2, 2]]
```

**Comment by zacw7** [comment]
> @zacw7 We need to provide specialization for primitive types (e.g. #5221)

Updated the code to skip the element-wise copying and use `add_items` instead which can add a fast path for primitives, as suggested by @laithsakka

**Pull request closed without merging**